Hmm, on Windows, the stage0 compiler doesn't contain Cargo?!
```
2023-10-22T00:04:07.2045142Z [2023-10-22T00:04:07.204Z INFO  opt_dist::tests] Using following `config.toml` for running tests:
2023-10-22T00:04:07.2045994Z     profile = "user"
2023-10-22T00:04:07.2046347Z     change-id = 115898
2023-10-22T00:04:07.2046679Z     
2023-10-22T00:04:07.2046963Z     [build]
2023-10-22T00:04:07.2047679Z     rustc = "C:/a/rust/rust/build/unpacked-dist/rustc-nightly-x86_64-pc-windows-msvc/rustc/bin/rustc.exe"
2023-10-22T00:04:07.2048362Z     
2023-10-22T00:04:07.2048599Z     [target.x86_64-pc-windows-msvc]
2023-10-22T00:04:07.2049156Z     llvm-config = "C:/a/rust/rust/build/x86_64-pc-windows-msvc/llvm/bin/llvm-config.exe"
2023-10-22T00:04:07.2049680Z     
2023-10-22T00:04:07.2051149Z [2023-10-22T00:04:07.204Z INFO  opt_dist::exec] Executing `COMPILETEST_FORCE_STAGE0=1 python C:\a\rust\rust\x.py test --stage 0 tests/assembly tests/codegen tests/codegen-units tests/incremental tests/mir-opt tests/pretty tests/run-pass-valgrind tests/ui --skip tests\codegen\vec-shrink-panik.rs [at C:\a\rust\rust]`
2023-10-22T00:04:07.3129497Z Traceback (most recent call last):
2023-10-22T00:04:07.3130313Z ##[group]Building bootstrap
2023-10-22T00:04:07.3130809Z   File "C:\a\rust\rust\x.py", line 50, in <module>
2023-10-22T00:04:07.3131309Z     bootstrap.main()
2023-10-22T00:04:07.3145733Z   File "C:\a\rust\rust\src\bootstrap\bootstrap.py", line 1134, in main
2023-10-22T00:04:07.3217062Z     bootstrap(args)
2023-10-22T00:04:07.3217696Z   File "C:\a\rust\rust\src\bootstrap\bootstrap.py", line 1101, in bootstrap
2023-10-22T00:04:07.3220138Z     build.build_bootstrap()
2023-10-22T00:04:07.3220795Z   File "C:\a\rust\rust\src\bootstrap\bootstrap.py", line 894, in build_bootstrap
2023-10-22T00:04:07.3222516Z     args = self.build_bootstrap_cmd(env)
2023-10-22T00:04:07.3222983Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-10-22T00:04:07.3223660Z   File "C:\a\rust\rust\src\bootstrap\bootstrap.py", line 967, in build_bootstrap_cmd
2023-10-22T00:04:07.3225305Z     raise Exception("no cargo executable found at `{}`".format(
2023-10-22T00:04:07.3226270Z Exception: no cargo executable found at `C:\a\rust\rust\build\x86_64-pc-windows-gnu\stage0\bin\cargo.exe`
```

--- a/src/tools/opt-dist/src/tests.rs
+++ b/src/tools/opt-dist/src/tests.rs
@@ -79,6 +79,8 @@ llvm-config = "{llvm_config}"
         env.python_binary(),
         x_py.as_str(),
         "test",
+        "--build",
+        env.host_triple(),
         "--stage",
         "0",
         "tests/assembly",

--- a/src/tools/opt-dist/src/tests.rs
+++ b/src/tools/opt-dist/src/tests.rs
@@ -24,11 +24,10 @@ pub fn run_tests(env: &Environment) -> anyhow::Result<()> {
     let host_triple = env.host_triple();
     let version = find_dist_version(&dist_dir)?;
 
-    // Extract rustc, libstd, cargo and src archives to create the optimized sysroot
+    // Extract rustc, libstd and src archives to create the optimized sysroot
     let rustc_dir = extract_dist_dir(&format!("rustc-{version}-{host_triple}"))?.join("rustc");
     let libstd_dir = extract_dist_dir(&format!("rust-std-{version}-{host_triple}"))?
         .join(format!("rust-std-{host_triple}"));
-    let cargo_dir = extract_dist_dir(&format!("cargo-{version}-{host_triple}"))?.join("cargo");
     let extracted_src_dir = extract_dist_dir(&format!("rust-src-{version}"))?.join("rust-src");
 
     // We need to manually copy libstd to the extracted rustc sysroot
@@ -47,8 +46,6 @@ pub fn run_tests(env: &Environment) -> anyhow::Result<()> {
 
     let rustc_path = rustc_dir.join("bin").join(format!("rustc{}", executable_extension()));
     assert!(rustc_path.is_file());
-    let cargo_path = cargo_dir.join("bin").join(format!("cargo{}", executable_extension()));
-    assert!(cargo_path.is_file());
 
     // Specify path to a LLVM config so that LLVM is not rebuilt.
     // It doesn't really matter which LLVM config we choose, because no sysroot will be compiled.
@@ -65,13 +62,11 @@ change-id = 115898
 
 [build]
 rustc = "{rustc}"
-cargo = "{cargo}"
 
 [target.{host_triple}]
 llvm-config = "{llvm_config}"
 "#,
         rustc = rustc_path.to_string().replace('\\', "/"),
-        cargo = cargo_path.to_string().replace('\\', "/"),
         llvm_config = llvm_config.to_string().replace('\\', "/")
     );
     log::info!("Using following `config.toml` for running tests:\n{config_content}");

